lib/test/Test/Pos/Txp/Toil/UtxoSpec.hs

```diff
@@ -456,3 +467,19 @@ txShouldFailWithPlutus res err = case res of
     other -> expectationFailure $
         "expected: Left ...: " <> show (WitnessScriptError err) <> "\n" <>
         " but got: " <> show other
+
+-- | Transaction should fail with a 'ToilInvalidWitness' specifically 'WitnessAdoptedScriptVerMismatch' error.
+txShouldFailWithInvalidWitness :: Either ToilVerFailure () -> Expectation
```

Comment:
I think it's better to make this name more specific, e. g. `txShouldFailWithInvalidAdoptedScriptVer`.